StackExchange/Stacks
# chore(button): design refinements

> **Note**
> Please don't merge before creating a plan to communicate the deprecation of muted+filled

---

[STACKS-339](https://stackoverflow.atlassian.net/browse/STACKS-339)

This PR implements button design refinements as part of the design vision changes to Stacks. cc @PiperLawson 

## How to test

Hit up [the buttons docs page in the deploy preview](https://deploy-preview-1421--stacks.netlify.app/product/components/buttons/). The buttons should match the Figma design spec at the end of this PR description. Test out combinations of light/dark and high contrast modes.

## TODO

- [x] Remove border on filled variations
- [x] Remove inset shadow on filled variations
- [x] Darken type of muted button styles
- [x] Lighten border color of muted button styles
- [x] Deprecate muted+filled button style
- [x] Ensure visual regression tests pass
- [ ] Create a plan to communicate the deprecation of muted+filled

## Details about changes in this PR

- You'll notice > 3,000 files changed. Nearly all are visual regression test baseline images, since we have > 1,000 images for button being tested against three different browsers
- A _deprecated_ warning has been added to the button docs for the muted+filled style and the exampled for that style in combination with loading and dropdown has been removed from the docs to discourage usage

## Figma

<img width="614" alt="image" src="https://github.com/StackExchange/Stacks/assets/647177/9cec535c-ed80-466b-ba3f-9c3959899d6d">
